design/one-pager-crossplane-state-metrics.md

```diff
@@ -0,0 +1,640 @@
+  target:
+    group: aws.platform.upbound.io
+    kind: XNetwork        # Supports wildcard "*" for all kinds
+    version: v1alpha1
+
+  # Optional: Filter resources
+  labelSelector: "team=platform"
+  fieldSelector: "metadata.namespace=production"
+
+  # Define metric values to extract (using JSONPath)
+  metrics:
+  - name: ready_status
+    path: "status.conditions[?(@.type=='Ready')].status"
+
+  # Define label dimensions (using JSONPath)
+  labels:
+  - name: resource_name
+    path: "metadata.name"
+  - name: region
+    path: "spec.parameters.region"
+  - name: team
+    path: "metadata.labels['team']"
+```
+
+**Key API Features:**
+
+1. **JSONPath Support**: Extract any field from the target resource using JSONPath expressions
+2. **Wildcard Kind Monitoring**: Use `kind: "*"` to monitor all resource types in a group/version
+3. **Flexible Filtering**: Label and field selectors to target specific resources
+4. **Configurable Intervals**: Per-metric reconciliation intervals to balance freshness vs. load
+5. **Automatic Metric Types**: Generates both gauge (current state) and counter (state transitions)
+
+#### Example: Namespace-Scoped Metric
+
+```yaml
+apiVersion: metrics.crossplane.io/v1alpha1
+kind: Metric
+metadata:
+  name: webapp-ready-status
+  namespace: team-a
+spec:
+  help: "WebApp ready status per instance"
+  interval: 1m
+  target:
+    group: platform.example.com
+    kind: WebApp
+    version: v1
+  metrics:
+  - name: ready_status
+    path: "status.conditions[?(@.type=='Ready')].status"
+  labels:
+  - name: app_name
+    path: "metadata.name"
+  - name: team
+    path: "metadata.labels['team']"
+  labelSelector: "environment=production"
+```
+
+#### Example: Wildcard Kind Monitoring
+
+Monitor all platform APIs with a single ClusterMetric:
+
+```yaml
+apiVersion: metrics.crossplane.io/v1alpha1
+kind: ClusterMetric
+metadata:
+  name: team-apis-ready-count
+spec:
+  help: "Ready status of all platform APIs (1=ready, 0=not ready)"
+  interval: 1m
+  target:
+    group: aws.platform.upbound.io
+    kind: "*"              # Monitors ALL kinds in this group
+    version: "*"           # Monitors ALL versions from this kinds
+  metrics:
+  - name: ready_status
+    path: "status.conditions[?(@.type=='Ready')].status"
+  labels:
+  - name: status
+    path: "status.conditions[?(@.type=='Ready')].status"
+  # Automatically adds 'kind' label: XNetwork, XVPC, XDatabase, etc.
+```
+
+### Metrics Generated
+
+For each ClusterMetric/Metric resource, the operator automatically generates **two metric types**:
+
+#### 1. Gauge Metrics (Current State)
+
+```promql
+# Pattern: crossplane_{metadata.name}
+crossplane_xnetwork_ready_status{resource_name="prod-vpc", region="us-west-2", network_id="vpc-123"} = 1
+
+# Provider health example
+crossplane_provider_health{provider_name="provider-aws", status="True"} = 1
+
+# Resource count example
+crossplane_xnetwork_resource_count{resource_name="prod-vpc", region="us-west-2"} = 15
```

Comment:
How is this configured? Is it created automatically for all monitored resources?